receiver/mysqlreceiver/client.go

```diff
@@ -182,6 +184,49 @@ func newMySQLClient(conf *Config) client {
 	}
 }
 
+func (c *mySQLClient) getInnodbStatusStats() (map[string]int64, error) {
+	/*
+		TODO: The NewInnodbStatusParser should be able to be created with the mySQLClient.
+	*/
+	innodbParser, err := parser.NewInnodbStatusParser()
+	if err != nil {
+		err := fmt.Errorf("could not create parser for innodb stats, %s", err)
+		return nil, err
+	}
+	var (
+		typeVar string
+		name    string
+		status  string
+	)
+
+	query := "SHOW /*!50000 ENGINE*/ INNODB STATUS;"
+	row := c.client.QueryRow(query)
+	mysqlErr := row.Scan(&typeVar, &name, &status)
+
+	// TODO: Suggest better value if there's an error for the metric.
+	if mysqlErr != nil {
+		err := fmt.Errorf("error querying the mysql db for innodb status")
+		return nil, err
+	}
+
+	innodbParser.SetInnodbStatusFromString(status)
+	metrics, errs := innodbParser.ParseStatus()
+
+	total_errs := 0
+	for key, val := range errs {
+		if errs[key][0] != nil {
+			fmt.Println(val)
```

Comment:
No `fmt.Println`. Either use a proper logging library or let caller handle displaying of error.

Reply:
Sure I'll change the print statements, to proper logging statements.

Reply:
Now we are returning all the errors, and let scraper handle error.